### PR TITLE
Endpoint for users to review all data at once in mephisto review

### PR DESCRIPTION
**Pull request by sghmk12**
This feature allows users to pull all data points from a review source at once to their front-end via our ```mephisto review``` server with a new endpoint with the URL ```/all_data```.

For example if the data source consisted the following ```data.csv``` file:
```
This is good text, row1
This is bad text, row2
```
The new endpoint will return the following response when queried:
response

To access this endpoint users must include a ```--all``` flag in their ```mephisto review``` command.
This can be in conjunction with the ```--db``` flag such as:
```
mephisto review my-review/build -o results.csv --db html-static-task-example --all
```
or in conjunction with standard input such as:
```
cat data.csv | mephisto review my-review/build -o results.csv --all
```

If the ```--all``` flag is not set the endpoint will return an error.
Conversely, If the ```--all``` flag is set and other endpoints for reviewing individual tasks are accessed they will also return error as a safety measure

Users can now query the endpoint ```/all_data``` for a list of all data points related to the task that they are currently reviewing

Users can also add parameters to the query such that they get a paginated result using the params ```page``` and ```results_per_page```

Such a query can look like this: ```/all_data?page=2&results_per_page=3```

If users do not specify a value for results per page the server will default to 10 results per page and if neither parameter is provided the server returns all results.

### CODE REVIEW

File: mephisto/client/review/review_server.py
Comment (pringshia) [inline mephisto/client/review/review_server.py:163]: I would just call this endpoint `/all_data`. The reason the other endpoint is called `/data_for_current_tasks` is bc it only works for a single task. having both all and current task in the endpoint name is sort of an oxymoron/contradictory.

File: mephisto/client/review/review_server.py
Comment (pringshia) [inline mephisto/client/review/review_server.py:101]: it might be cleaner to replace these Lines 92-101 here with this, using a default parameter value & assertion, e.g.:

```python
def consume_all_data(page, results_per_page = RESULTS_PER_PAGE_DEFAULT):
    assert type(results_per_page) is int, "results_per_page should be an int"
```

File: mephisto/client/review/review_server.py
Comment (pringshia) [inline mephisto/client/review/review_server.py:103]: you don't need to do a None check if you're also doing isinstance, you can just do the isinstance, since it will also cover the None clause (a value of None will always make isinstance false, i.e. `isinstance(None, int) == false`)

that is to say you could just shorted all of the above with:
```python
paginated = type(page) is int and page > 0
```

File: mephisto/client/review/review_server.py
Comment (pringshia) [inline mephisto/client/review/review_server.py:92]: is page 0-indexed or 1-indexed? a comment for this method may be useful

File: mephisto/client/review/review_server.py
Comment (pringshia) [inline mephisto/client/review/review_server.py:92]: You may want to write this comment as a [Python docstring](https://www.programiz.com/python-programming/docstrings) instead, such that IDEs can pick up and surface documentation for it.

**Comment by pringshia** [comment]
Overall this looks good! One minor comment about using a docstring. Aside from that, approved on my end.

@JackUrb since this PR is quite Python-heavy do you mind taking a quick scan? I tried my best on any Python feedback I could give.

@sghmk12 for this one since it's Python-heavy, let's also wait for an approval from Jack before merging in

**Review by pringshia — APPROVED** [review]

**Review by JackUrb — CHANGES_REQUESTED** [review]
Sorry about the late review! Some comments inline, but the basic overview is that, while this method will _work_, on larger review loads it's going to be slow. In short, calls to the `get_units_for_task_name` and `get_data_from_unit` methods are going to be a bottleneck, and should optimally be minimized.

The first major change is to treat the `MephistoDataBrowser` data in the same way as local data, populating the `all_data_list` on the first initialization.

Then, in order to pull the strongest benefits from pulling from the DB directly, you can implement one of two ways to refresh the `all_data_list` from the database, first defining a `fill_from_db` method or similar, and then calling it either periodically from a thread, or after enough time has passed since the last call during an `all_task_data` request.

Sorry about the late context here, but great work otherwise!

File: mephisto/client/review/review_server.py
Comment (sghmk12) [reply mephisto/client/review/review_server.py]: Thanks for the input, definitely much needed advice. Just implemented these features. Please let me know if more is needed or if there are any python errors in my latest commits!

File: mephisto/client/review/review_server.py
Comment (pringshia) [inline mephisto/client/review/review_server.py:227]: maybe I'm missing something, but the above 3 lines here could just be `all_data_list = list(data_source)` no?

Comment (sghmk12) [reply mephisto/client/review/review_server.py:227]: You are correct, I am still getting accustomed to advanced python syntax haha. Will change

File: mephisto/client/review/review_server.py
Comment (JackUrb) [inline mephisto/client/review/review_server.py:110]: This (5) should be moved to a constant near the top of the file for easy editing. Generally it's good to avoid magic numbers (could also move both by making the constant based on a second timeout).

**Review by JackUrb — APPROVED** [review]
Great changes - sorry for the delay! Just one nit but otherwise this is very much good to go.

File: mephisto/client/review/review_server.py
Comment (pringshia) [reply mephisto/client/review/review_server.py:110]: It would also be good to have a boolean constant flag to toggle this functionality on/off altogether

Comment (sghmk12) [reply mephisto/client/review/review_server.py:110]: Added both for my next PR. My bad I didn't have this in the current version.

**Pull request merged**
